### PR TITLE
[release-0.17] Prepare v0.17 release

### PR DESCRIPTION
**Pull request by marquiz**
Generated with:

    ./hack/prepare-release.sh -g 1.23.4 v0.17.0

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,7 +8,7 @@ IMAGE_BUILD_CMD ?= docker build
 IMAGE_BUILD_EXTRA_OPTS ?=
 IMAGE_PUSH_CMD ?= docker push
 CONTAINER_RUN_CMD ?= docker run
-BUILDER_IMAGE ?= golang:1.23-bookworm
+BUILDER_IMAGE ?= golang:v1.23.4-bookworm
```
Comment (TessaIO) [inline Makefile:11]:
```suggestion
BUILDER_IMAGE ?= golang:1.23.4-bookworm
```

File: README.md [hunk 2]
```diff
@@ -48,4 +48,4 @@ $ kubectl get no -o json | jq ".items[].metadata.labels"
 
 ```
 
-[documentation]: https://kubernetes-sigs.github.io/node-feature-discovery
+[documentation]: https://kubernetes-sigs.github.io/node-feature-discovery/v0.17
```
Comment (TessaIO) [inline README.md:51]: I wonder if we should include the patch version in URL also. i.e v0.17.x. Does it make sense?

File: Makefile (hunk 1 above)
Comment (marquiz) [reply Makefile]: :man_facepalming: thanks for noticing this, fixed

File: README.md (hunk 2 above)
Comment (marquiz) [reply README.md:51]: Good question. The documentation is arranged per release branch. Users always see the documentation of the latest patch release. What you suggested is not supported by the infra.

We don't make available or preserve documentation for each patch release. We could, if there's value for that but that's another story

**Comment by marquiz** [comment]
Wonder why the master branch linger there

/override pull-node-feature-discovery-build pull-node-feature-discovery-build-gh-pages pull-node-feature-discovery-build-image  pull-node-feature-discovery-verify

**Comment by marquiz** [comment]
@TessaIO PTAL

**Pull request merged**